### PR TITLE
Sso login

### PR DESCRIPTION
**Pull request by vikramdevtron**

### CODE REVIEW

File: pkg/sso/SSOLoginService.go [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+		data := cm.Data
+		data["dex.config"] = updatedData["dex.config"]
+		cm.Data = data
+		_, err = impl.K8sUtil.UpdateConfigMapFast(ArgocdConfigMapNamespace, cm, client)
+		if err != nil {
+			impl.logger.Warnw(" config map failed", "err", err)
+			continue
+		}
+		if err == nil {
+			impl.logger.Warnw(" config map apply succeeded", "on retryCount", retryCount)
```
Comment (nishant-d) [inline pkg/sso/SSOLoginService.go:155]: is it warn or info?

File: pkg/sso/SSOLoginService.go [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+	configDataByte, err := json.Marshal(request.Config)
+	if err != nil {
+		return nil, err
+	}
+	model, err := impl.ssoLoginRepository.GetById(request.Id)
+	if err != nil {
+		impl.logger.Errorw("error in update new sso login config", "error", err)
+		return nil, err
+	}
+	model.Url = request.Url
```
Comment (nishant-d) [inline pkg/sso/SSOLoginService.go:190]: code duplicacy from previous method

**Review by nishant-d — CHANGES_REQUESTED** [review]

File: pkg/sso/SSOLoginService.go [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+		ssoLoginRepository:  ssoLoginRepository,
+		K8sUtil:             K8sUtil,
+		clusterService:      clusterService,
+		envService:          envService,
+	}
+	return serviceImpl
+}
+
+const ArgocdConfigMap string = "argocd-cm"
+const ArgocdConfigMapNamespace string = "devtroncd"
```
[the remark this answers is not in the document]
Comment (vikramdevtron) [reply pkg/sso/SSOLoginService.go:74]: > hardcoding

should we move it to env variables. ?

File: pkg/sso/SSOLoginService.go (hunk 1 above)
Comment (vikramdevtron) [reply pkg/sso/SSOLoginService.go:155]: > is it warn or info?

I think Debugw will be fine there.

File: pkg/sso/SSOLoginService.go (hunk 2 above)
Comment (vikramdevtron) [reply pkg/sso/SSOLoginService.go:190]: > code duplicacy from previous method

create and update share some same piece code, yes separated common func requred for that.

**Review by nishant-d — APPROVED** [review]

**Pull request merged**